sklearn/utils/_test_common/instance_generator.py

```diff
@@ -590,6 +590,16 @@
         ],
     },
     MDS: {"check_dict_unchanged": dict(max_iter=5, n_components=1, n_init=2)},
+    MLPClassifier: {
+        "check_sample_weight_equivalence": [
+            dict(solver="lbfgs"),
+        ]
+    },
+    MLPRegressor: {
+        "check_sample_weight_equivalence": [
+            dict(solver="sgd", tol=1e-2, random_state=42),
+        ]
+    },
```

Comment:
We might need similar entries for `check_sample_weight_equivalence_on_sparse_data`.


EDIT: I tried locally, and this check passes with the default settings (lbfgs). I think testing the sgd solver on dense data is enough.